File: py-server/app/routes/user_routes.py
from flask import Blueprint
from app.controllers import user_controller as uc

bp = Blueprint('user_routes', __name__)


@bp.route('/users', methods=['GET'])
def get_users_route():
    return uc.get_users()


@bp.route('/users/<int:user_id>', methods=['GET'])
def get_user_route(user_id):
    return uc.get_user(user_id)


@bp.route('/users', methods=['POST'])
def post_user_route():
    return uc.add_user()


@bp.route('/users/<int:user_id>', methods=['PATCH'])
def patch_user_route(user_id):
    return uc.update_user(user_id)


@bp.route('/users/<int:user_id>/facial-recognition', methods=['PATCH'])
def patch_facial_recognition_route(user_id):
    return uc.update_facial_recognition(user_id)


@bp.route('/users/<int:user_id>/image-key', methods=['PATCH'])
def patch_image_key_route(user_id):
    return uc.update_image_key(user_id)


@bp.route('/users/<int:user_id>', methods=['DELETE'])
def delete_user_route(user_id):
    return uc.delete_user(user_id)

File: js-server/routes/user_routes.js
const express = require('express');
const router = express.Router();
const rutasUsuario =  require('../controllers/user_controller')
const multer = require('multer');
const upload = multer({ storage: multer.memoryStorage() }); // Configuración para manejar archivos



//mostrar usuarios
router.get('/users',rutasUsuario.get_users)
//mostrar usuario por su id
router.get('/users/:user_id',rutasUsuario.get_user)
//registrar usuario
router.post('/users', upload.single('image'), rutasUsuario.addUser);
//actualizar usuario
router.patch('/users/:user_id',upload.single('image'),rutasUsuario.update_user);
//eliminar usuario
router.delete('/users/:user_id', rutasUsuario.delete_user_route);
//agregar imagen clave para el reconocimiento facial
router.patch('/users/:user_id/facial-recognition',upload.single('image_key'),rutasUsuario.update_facial_recognition);
//modificar imagen clave para el reconocimiento facial
router.patch('/users/:user_id/facial-recognition',upload.single('image_key'),rutasUsuario.update_image_key);

module.exports = router;